### PR TITLE
cancel queue request on signal to prevent streams from blocking

### LINKED ISSUE #45: Add script-wrapper to manage consumers via PCNTL messages
Idle consumers currently cannot be stopped from the CLI using system signals. This is due to `queue->consume` being a blocking call. SO `pcntl_signal_dispatch` does not get called to terminate the script. 

Idea would be to create a wrapper script around the actual `start consumer` script that listens to the system signals instead and turns interrupting signals into a shutdown message sent to the consumer's queue. This would shut down the consumer right away (as the queue is empty). 

The signal itself still would need to be send to the consumer directly as well to handle situations where the queue is not empty (and hence the shutdown messages is further down the queue). This can be quite tricky to handle as this could potentially lead to situations where the previously send shutdown message is received later once the consumer is restarted.

### PR DESCRIPTION
**Pull request by oqq**
The amqp extension + library has its own mechanism to remove blocking for requests, if the signal handler is activated. 

But the cancelation for requests was not called from this library, so i have moved some code around to get rid of that issue: closes #45 

I am not sure about side effects, so please have an extra review round. ;)

### CODE REVIEW

**Comment by prolic** [comment]
@oqq see https://github.com/oqq/HumusAmqp/blob/0e1b18f0e8ae27aeb587f64b1653b9d39ef29217/src/AbstractConsumer.php#L182

this line is 100% blocking, so the shutdown method is never called, unless the block has been removed (f.e. by timeout or message came in)

**Comment by oqq** [comment]
This is not true for the php-amqplib package.
See this File: https://github.com/php-amqplib/php-amqplib/blob/master/PhpAmqpLib/Wire/IO/StreamIO.php and search for `canDispatchPcntlSignal`.

How I said, this package handles this behavior already like a charm by endless looping requests without blocking. My provided patch works exactly how it should without any script wrapper. Please test it by your own.

File: src/Driver/PhpAmqpLib/Queue.php [hunk 1]
```diff
@@ -304,7 +304,7 @@ public function cancel(string $consumerTag = '')
         $this->channel->getResource()->basic_cancel(
             $consumerTag,
             (bool) ($this->flags & Constants::AMQP_NOWAIT),
-            false
+            true
```
Comment (prolic) [inline src/Driver/PhpAmqpLib/Queue.php]: why is this change needed? you can control this behaviour this the AMQP_NOWAIT flag.

Comment (prolic) [reply src/Driver/PhpAmqpLib/Queue.php]: ping @oqq

Comment (oqq) [reply src/Driver/PhpAmqpLib/Queue.php]: But is AMQP_NOWAIT always set?

How I have tested this, basic_cancel must called with ($tag, false, true) if the server should be stopped. Maybe we add another method "stop" here and hard set (false, true)?

**Comment by prolic** [comment]
manually merged into master
